subsys/net/ip/Kconfig

```diff
@@ -56,6 +56,9 @@ source "subsys/net/ip/Kconfig.ipv4"
 config NET_SHELL
 	bool "Enable network shell utilities"
 	select SHELL
+	select NET_SOCKETS if DNS_RESOLVER
+	select NEWLIB_LIBC if DNS_RESOLVER
+	select NET_SOCKETS_POSIX_NAMES if DNS_RESOLVER
```

Comment:
Regardless of other matters, NET_SOCKETS_POSIX_NAMES should not be used by any internal code in Zephyr itself. It's strictly a compatibility option for external applications of a particular type. It's not compatible with POSIX subsystem (soon there will conflicting anti-depend on it).

Internal Zephyr code should always use native Zephyr API, which is zsock_getaddrinfo().